app/src/main/java/com/assist_software/expenseappmvp/screens/registerScreen/RegisterPresenter.kt

```diff
@@ -34,17 +53,77 @@ class RegisterPresenter(
     }
 
     private fun registerUser(): Disposable {
-        return view.registerUser()
-            .observeOn(rxSchedulers.background())
-            .flatMap {
-                userRepository.savePrimaryUser(user).toObservable()
+        return view.registerUserClick()
+            .observeOn(rxSchedulers.androidUI())
+            .doOnNext { fieldValidation() }
+            .filter { !(user.userName == "" || user.userEmail == "" || user.userPassword == "") }
+            .doOnNext {
```

Comment:
At register it enters on this doOnNext and the applications does nothing because there is no data at this point